### PR TITLE
Dropping support for symfony < 3.4

### PR DESCRIPTION
**Pull request by franmomu**
## Subject

This is the same thing as https://github.com/sonata-project/SonataAdminBundle/pull/5733, dropping support of Symfony < 3.4 and >= 4, < 4.2

I am targeting this branch, because these changes respect BC.

## Changelog

```markdown
### Removed
- Support for Symfony < 3.4
- Support for Symfony >= 4, < 4.2
```

There are some changes that weren't marked as deprecated:

- [Removing `getName` from Form](https://github.com/symfony/symfony/blob/3.4/UPGRADE-3.0.md#form)

### CODE REVIEW

**Comment by core23** [comment]
Thanks @franmomu

**Comment by franmomu** [comment]
> Btw did you check for some @Legacy tests?
> 
> Same for the other PRs

mmmm I'll take a look, I basically checked for:
- sf\s?\d
- sf\s?<
- symfony\s?\d
- symfony\s?<
- \d\+\W
- drop
- bump
- version_compare
- next_major

and maybe something else that I don't remember right now

**Comment by OskarStark** [comment]
method_exists, class_exists, @group legacy

File: src/Form/Type/CommentType.php [hunk 1]
```diff
@@ -48,11 +48,6 @@ public function getBlockPrefix()
         return 'sonata_post_comment';
     }
 
-    public function getName()
```
Comment (phansys) [inline src/Form/Type/CommentType.php]: This is a BC break.

**Pull request merged**